src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -270,10 +270,13 @@ public static SolutionStateChecksums Deserialize(ObjectReader reader)
                 if (projectCone != null && !projectCone.Contains(projectId))
                     continue;
 
-                if (projectState.TryGetStateChecksums(out var projectStateChecksums) &&
-                    searchingChecksumsLeft.Remove(projectStateChecksums.Checksum))
+                if (projectState.TryGetStateChecksums(out var projectStateChecksums))
                 {
-                    result[projectStateChecksums.Checksum] = projectStateChecksums;
+                    if (searchingChecksumsLeft.Remove(projectStateChecksums.Checksum))
+                        result[projectStateChecksums.Checksum] = projectStateChecksums;
+
+                    if (searchingChecksumsLeft.Remove(projectStateChecksums.Info))
```

Comment:
>if (searchingChecksumsLeft.Remove(projectStateChecksums.Info)) [](http://example.com/codeflow?start=20&length=62)

This feels like a bug fix, but I'm curious whether it's an optimization and I'm missing something

Reply:
Oh, I see. The new GetAssetsAsync call you added for the optimization uses the Info with the TopLevelProjects path.